hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -0,0 +1,30 @@
+package org.apache.hudi.utilities.transform;
+
+import org.apache.hudi.common.util.TypedProperties;
+
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * An abstract {@link Transformer} to chain other {@link Transformer}s and apply sequentially.
+ * <p>
+ * A subclass is to supply a {@link List} of {@link Transformer}s in desired sequence.
+ */
+public abstract class ChainedTransformer implements Transformer {
+
+  protected abstract Supplier<List<Transformer>> supplyTransformers();
+
+  @Override
+  public Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties) {
```

Comment:
It would be better to have a test case which extends `ChainedTransformer ` and mock some transformer to verify this function.